src/Raven.Server/Documents/PeriodicBackup/RavenFtpClient.cs

```
@@ -65,135 +67,121 @@ public void UploadFile(string folderName, string fileName, Stream stream)
-            using (request.GetResponse())
-            {
-                Progress?.UploadProgress.ChangeState(UploadState.Done);
+                Progress?.UploadProgress.ChangeState(UploadState.PendingResponse);
+                if (client.FileExists(url))
+                {
+                    Progress?.UploadProgress.ChangeState(UploadState.Done);
+                }
             }
         }
 
         private string CreateNestedFoldersIfNeeded(string folderName)
         {
-            ExtractUrlAndDirectories(out var url, out var directories);
+            ExtractUrlAndDirectories(out var directories);
 
             // create the nested folders including the new folder
             directories.Add(folderName);
+            string url = "";
 
             foreach (var directory in directories)
             {
                 if (directory == string.Empty)
                     continue;
 
                 url += $"/{directory}";
-                var request = CreateFtpWebRequest(url, WebRequestMethods.Ftp.MakeDirectory, keepAlive: false);
-
-                try
-                {
-                    var response = request.GetResponse();
-                    response.Close();
-                }
-                catch (WebException e)
+                using (var client = CreateFtpClient(_url, keepAlive: false))
                 {
-                    var response = (FtpWebResponse)e.Response;
-                    if (response.StatusCode == FtpStatusCode.ActionNotTakenFileUnavailable)
+                    try
                     {
-                        // folder already exists
-                        continue;
+                        if (!client.DirectoryExists($"/{directory}"))
+                            client.CreateDirectory($"/{directory}");
+                    }
+                    catch (Exception e)
+                    {
+                        throw new Exception();
```

lets throw the original exception here

---

Done